Memorize/MemoryGame.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  MemoryGame.swift
+//  Memorize
+//
+//  Created by Kedar Dhere on 8/10/22.
+//
+
+import Foundation
+
+struct MemoryGame<CardContent> where CardContent: Equatable {
+    // MARK: - Properties
+    
+    struct Card: Identifiable {
+        var isFaceUp: Bool = false
+        var isMatched: Bool = false
+        var content: CardContent
+        var isPreviouslySeen: Bool = false
+        var id: Int
+    }
+    private(set) var cards: [Card]
+    private var indexOfTheOneAndOnlyFaceUpCard: Card?
+    private(set) var score: Int = 0
+    
+    // MARK: - Initilization
+    
+    init(numberOfPairsOfCards: Int, createCardContent: (Int) -> CardContent) {
+        cards = [Card]()
+        for pairIndex in 0..<numberOfPairsOfCards {
+            let content: CardContent = createCardContent(pairIndex)
+            cards.append(Card(content: content, id: pairIndex*2))
+            cards.append(Card(content: content, id: pairIndex*2 + 1))
+        }
+        cards.shuffle()
+    }
+    
+    // MARK: - Functions
+    
+    /**
+     This function is used to update the score and card details to statisy the following game conditions: 1. At a time only two cards should be faceUp.
+     2. If the content of the card is matching then update the score by two.
+     3. else facedown all the cards and if the if the any one of the card is already seen then reduce score by one.
+    */
```

Comment:
Replace with `///`.  It's Apple's current convention.  (The Javadoc-style docstring is dated for Swift.)